lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -460,6 +496,117 @@ def _check_job_arguments(self, jobs, has_parent_job=False):
+        # Get run job params from db, and inject parent job id, then run it
+        run_job_params = self._get_run_job_params_from_existing_job(
+            job, user_id=self.sdkmr.user_id
+        )
+        run_job_params[_PARENT_RETRY_JOB_ID] = job_id
+        # Submit job to job scheduler or fail and not count it as a retry attempt
+        child_job_id = self.run(params=run_job_params, as_admin=as_admin)
+
+        # TODO verify only parents have a batch_job, maybe rename to batch_job_parent?
+        if run_job_params.get("batch_job"):
+            # Otherwise we have to make this endpoint always return a list of retried_jobs?
+            raise Exception("Use batch retry endpoint for this job. ")
+
+        # Save that the job has been retried, and increment the count
+
+        if job.retried:
+            job.retry_count += 1
+        else:
+            job.retried = True
+            job.retry_count = 1
+
+        job.save()
+        # Should we compare the original and child job to make sure certain fields match,
+        # to make sure the retried job is correctly submitted? Or save that for a unit test?
+        return child_job_id
+
+    @staticmethod
+    def _get_job_input_params_from_existing_job(job_input: JobInput) -> Dict:
+        """
+        Get input level fields.
+        """
+        # Expected fields are found in `_init_job_rec`
+        inputs = dict()
+
+        # Used to iterate over "Job" attributes but in dictionary like lookup
+        inputs_list = [
+            _METHOD,
+            _APP_PARAMS,
+            _SERVICE_VER,
+            _APP_ID,
+            _PARENT_JOB_ID,
+            _WORKSPACE_ID,
+            "requirements",
+        ]
+
+        # Insert field if available
+        for field in inputs_list:
+            if job_input[field]:
+                inputs[field] = job_input[field]
+
+        # _SOURCE_WS_OBJECTS list
+        if job_input[_SOURCE_WS_OBJECTS]:
+            inputs[_SOURCE_WS_OBJECTS] = list(job_input[_SOURCE_WS_OBJECTS])
+
+        # Special Cases: It is OK if meta/narr_cell_info is in params 2x
+        if job_input["narrative_cell_info"]:
+            # For easier testing
+            nci = job_input["narrative_cell_info"]  # type: Meta
+            inputs["meta"] = inputs["narrative_cell_info"] = nci.to_mongo().to_dict()
+
+        # Require a non blank params for job browser or narrative compatibility
+        if _APP_PARAMS not in inputs:
+            inputs[_APP_PARAMS] = {}
+
+        return inputs
+
+    @staticmethod
+    def _get_run_job_params_from_existing_job(job: Job, user_id: str) -> Dict:
+        """
+        Get top level fields from job model to be sent into `run_job`
+        """
+        ji = job.job_input  # type: JobInput
+        job_input = EE2RunJob._get_job_input_params_from_existing_job(ji)
+
+        run_job_params = {
+            _WORKSPACE_ID: job.wsid,  # optional
+            _NARRATIVE_CELL_INFO: job_input.get(_NARRATIVE_CELL_INFO),  # optional
```

Comment:
This isn't a field in RunJobParams AFAICT

Reply:
Removed _NARRATIVE_CELL_INFO in next PR

Reply:
:+1: